src/main/java/pl/krakow/controller/CustomerController.java

```diff
@@ -0,0 +1,70 @@
+package pl.krakow.controller;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+import pl.krakow.domain.Customer;
+import pl.krakow.exception.CustomerExistException;
+import pl.krakow.exception.CustomerNotFoundException;
+import pl.krakow.service.CustomerRepository;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/customers")
+public class CustomerController {
+
+    private CustomerRepository customerRepository;
+
+    public CustomerController(CustomerRepository customerRepository) {
+        this.customerRepository = customerRepository;
+    }
+
+    @GetMapping(path = { "", "/" })
+    public List<Customer> getAllCustomers() {
+        return customerRepository.findAll();
+    }
+
+    @GetMapping(path = "/{name}")
+    public Customer getCustomerByName(@PathVariable("name") String name) {
+        try {
+            return customerRepository.findByName(name);
+        }
+        catch (CustomerNotFoundException exception) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND);
+        }
+    }
+
+    @PostMapping(path = { "", "/" }, consumes = "application/json")
```

Comment:
I would move such literals to constants.

Reply:
done!